staging/src/k8s.io/client-go/transport/transport.go

```diff
@@ -204,6 +204,20 @@ func TLSConfigFor(c *Config) (*tls.Config, error) {
 		}
 	}
 
+	// Add support for writing TLS client random and master secret in NSS key log format.
+	// https: //developer.mozilla.org/en-US/docs/Mozilla/Projects/NSS/Key_Log_Format
+	// Only available in debug mode.
+	if klog.V(9).Enabled() { // nolint:logcheck
```

Comment:
> Key logs can be written by NSS so that external programs can decrypt TLS connections.

I'm *very* squeamish about writing out anything that can be used to compromise TLS connections. We've had issues in the past where stuff written with debug log levels was considered a CVE and a disclosure (like tokens or credentials).

Reply:
oh, that is bad, I was assuming that since this is implemented, in curl, chrome and firefox and all these libraries https://everything.curl.dev/usingcurl/tls/sslkeylogfile.html#restrictions was "more common"

Reply:
it seems is being standardized and has a chapter about it https://datatracker.ietf.org/doc/html/draft-thomson-tls-keylogfile-01#section-3

Reply:
ok, no big deal, better to not take the risk, it seems the security angle depend on the protection of the exported file, but I can see how people can be creative and inject env variables and dump into some shared mount or something

/close